stackslib/src/chainstate/stacks/boot/signers-voting.clar

```diff
@@ -3,27 +3,42 @@
 ;;
 
 ;; maps dkg round and signer to proposed aggregate public key
-(define-map votes {reward-cycle: uint, round: uint, signer: principal} {aggregate-public-key: (buff 33), reward-slots: uint})
+(define-map votes {reward-cycle: uint, round: uint, signer: principal} {aggregate-public-key: (buff 33), signer-weight: uint})
 ;; maps dkg round and aggregate public key to weights of signers supporting this key so far
 (define-map tally {reward-cycle: uint, round: uint, aggregate-public-key: (buff 33)} uint)
-;; maps aggregate public keys to rewards cycles and rounds
-(define-map used-aggregate-public-keys (buff 33) {reward-cycle: uint, round: uint})
-
-(define-constant err-signer-index-mismatch (err u10000))
-(define-constant err-invalid-signer-index (err u10001))
-(define-constant err-out-of-voting-window (err u10002))
-(define-constant err-old-round (err u10003))
-(define-constant err-ill-formed-aggregate-public-key (err u10004))
-(define-constant err-duplicate-aggregate-public-key (err u10005))
-(define-constant err-duplicate-vote (err u10006))
-(define-constant err-invalid-burn-block-height (err u10007))
+;; maps aggregate public keys to rewards cycles
+(define-map used-aggregate-public-keys (buff 33) uint)
+
+;; Error codes
+;; 1 - 9 are reserved for use in the .signers contract, which can be returned
+;; through this contract)
+(define-constant ERR_SIGNER_INDEX_MISMATCH u10)
+(define-constant ERR_INVALID_SIGNER_INDEX u11)
+(define-constant ERR_OUT_OF_VOTING_WINDOW u12)
+(define-constant ERR_ILL_FORMED_AGGREGATE_PUBLIC_KEY u13)
+(define-constant ERR_DUPLICATE_AGGREGATE_PUBLIC_KEY u14)
+(define-constant ERR_DUPLICATE_VOTE u15)
+(define-constant ERR_FAILED_TO_RETRIEVE_SIGNERS u16)
+(define-constant ERR_INVALID_ROUND u17)
 
 (define-constant pox-info
     (unwrap-panic (contract-call? .pox-4 get-pox-info)))
 
-;; maps reward-cycle ids to last round
+;; Threshold consensus, expressed as parts-per-thousand to allow for integer
+;; division with higher precision (e.g. 700 for 70%).
+(define-constant threshold-consensus u700)
+
+;; Maps reward-cycle ids to last round
 (define-map rounds uint uint)
 
+;; Maps reward-cycle ids to aggregate public key.
+(define-map aggregate-public-keys uint (buff 33))
```

Comment:
Great job on importing the tests from #4392, going to close accordingly.

From writing the tests^ I believe everything LGTM, the *only* thing I can point out atm that we may need to change but isn't breaking anything is that we now have two maps that track cycle-id to aggregate public key (here & in PoX-4.clar).

The latter map currently isn't written into or fetched so probably best to just remove from PoX-4.clar & keep the one in here.

Reply:
Good call. Deleted in a92dbdb83.